dali/pipeline/operators/crop/crop.h

```diff
@@ -1,176 +1,188 @@
-    } else if (output_type_ == DALI_INT32) {
-      RunHelper<int>(ws, idx);
-    } else if (output_type_ == DALI_INT64) {
-      RunHelper<int64>(ws, idx);
-    } else {
-      DALI_FAIL("Unsupported output type.");
-    }
-  }
-
-  Tensor<CPUBackend> input_ptrs_, input_strides_;
-  Tensor<GPUBackend> input_ptrs_gpu_, input_strides_gpu_;
-  vector<int> crop_offsets_;
-
-  std::vector<std::pair<int, int>> per_sample_crop_;
-  std::vector<std::pair<int, int>> per_sample_dimensions_;
-
- protected:
-  // Output data type
-  DALIDataType output_type_;
-
-  // Output data layout
-  DALITensorLayout output_layout_;
-
-  USE_OPERATOR_MEMBERS();
-};
-
-}  // namespace dali
-
-#endif  // DALI_PIPELINE_OPERATORS_CROP_CROP_H_
-
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_CROP_CROP_H_
+#define DALI_PIPELINE_OPERATORS_CROP_CROP_H_
+
+#include <vector>
+#include <utility>
+
+#include "dali/common.h"
+#include "dali/error_handling.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+
+namespace dali {
+
+class CropAttr {
+ protected:
+  explicit inline CropAttr(const OpSpec &spec)
+      : image_type_(spec.GetArgument<DALIImageType>("image_type")),
+        C_(IsColor(image_type_) ? 3 : 1),
+        batch_size_{spec.GetArgument<int>("batch_size")} {
+    if (spec.name() != "Resize") {
+      vector<float> cropArgs = spec.GetRepeatedArgument<float>("crop");
+
+      DALI_ENFORCE(cropArgs[0] >=0,
+        "Crop height must be greater than zero. Received: " + std::to_string(cropArgs[0]));
+      DALI_ENFORCE(cropArgs[1] >=0,
+        "Crop width must be greater than zero. Received: " + std::to_string(cropArgs[0]));
+
+      crop_height_ = std::vector<int>(batch_size_, static_cast<int>(cropArgs[0]));
+      crop_width_ = std::vector<int>(batch_size_, static_cast<int>(cropArgs[1]));
+
+      crop_x_norm_ = std::vector<float>(batch_size_, -1.0f);
+      crop_y_norm_ = std::vector<float>(batch_size_, -1.0f);
+    }
+  }
+
+  std::pair<int, int> SetCropXY(const OpSpec &spec, const ArgumentWorkspace *ws,
+                                const Index dataIdx, int H, int W) {
+    DALI_ENFORCE(H >= crop_height_[dataIdx]);
+    DALI_ENFORCE(W >= crop_width_[dataIdx]);
+
+    auto crop_x_norm =
+        std::vector<float>(batch_size_, spec.GetArgument<float>("crop_pos_x", ws, dataIdx));

```

Comment:
We create here a vector initialized to crop_pos_x, just to access dataIdx in that vector (and we can simply use crop_pos_x), and than we discard those vectors after the call.